relays/ethereum-client/src/client.rs

```diff
@@ -28,22 +28,35 @@ use jsonrpsee::Client as RpcClient;
 /// The client used to interact with an Ethereum node through RPC.
 #[derive(Clone)]
 pub struct Client {
+	params: ConnectionParams,
 	client: RpcClient,
 }
 
 impl Client {
 	/// Create a new Ethereum RPC Client.
 	pub fn new(params: ConnectionParams) -> Self {
+		Self {
+			client: Self::build_client(&params),
+			params,
+		}
+	}
+
+	/// Build client to use in connection.
+	fn build_client(params: &ConnectionParams) -> RpcClient {
 		let uri = format!("http://{}:{}", params.host, params.port);
 		let transport = HttpTransportClient::new(&uri);
 		let raw_client = RawClient::new(transport);
-		let client: RpcClient = raw_client.into();
-
-		Self { client }
+		raw_client.into()
 	}
 }
 
 impl Client {
+	/// Reopen client connection.
+	pub fn reconnect(mut self) -> Self {
+		self.client = Self::build_client(&self.params);
```

Comment:
Yes - at least in our version of `jsonrpsee`. Maybe in v2 something has changed, but in v1 background thread was just stopped in case of error => client should have been restarted